### PR TITLE
Support pandas objects in merge

### PR DESCRIPTION
**Pull request by phofl**

### CODE REVIEW

File: dask_expr/_collection.py [hunk 1]
```diff
@@ -1316,17 +1283,56 @@ def merge(
...
+
+    if not is_dask_collection(right):
+        if left_index and right_on:  # change to join on index
+            right = right.set_index(right[right_on])
+            right_on = None
+            right_index = True
+        right = from_pandas(right, npartitions=1)
+
+    left = left.expr
+    right = right.expr if isinstance(right, FrameBase) else from_pandas(right, 1).expr
```
Comment (rjzamora) [inline dask_expr/_collection.py:1320]:
```suggestion
    right = right.expr
```
Isn't it now impossible for `isinstance(right, FrameBase)` to return `True` now that you have added the `is_dask_collection` check above?

**Review by rjzamora — COMMENTED** [review]
Where should the doc-string live in cases like this? Do we move it to the stand-alone `merge` function, or leave it in the `merge` method? Either way, we probably want to clarify that `left`/`right` may be a `pd.DataFrame` object.

File: dask_expr/_collection.py (hunk 1 above)
Comment (phofl) [reply dask_expr/_collection.py]: Good point, thx

**Comment by phofl** [comment]
Hm that's a good question, it would put it onto the collection, but no strong opionion

pandas uses something like shared docstrings for cases like this, but I don't like them at all

**Pull request merged**